### DISCUSSION TITLE
Add support for Polar.sh

### DISCUSSION THREAD

**Discussion by mikker** [pay-rails/pay#1128] [Ideas]
Hey!

Just checking in to report that I've begun the task of adding support for [Polar.sh](https://polar.sh).

They're fairly new on the scene but the team is awesome and very helpful and they look like a very capable competitor, especially by being Merchant of Record, supporting both subscriptions and one-time payments, and having all sorts of integrations that're as easy as click-to-enable.

I'll make a proper PR when I'm nearing anything presentable but until then, wish me luck 😅

**Comment by excid3** [comment]
Polar looks cool!

I don't think I'll have time to officially maintain, but how making it as a plugin and we can link to it as a community maintained integration?

**Reply by excid3** [reply]
Also let me know if you have any trouble and we can make improvements to Pay to make adding new payment processors easier. 👍

**Comment by mikker** [comment]
That's totally fair. If I get so far, I'm also willing to have the responsibility to maintain the polar bits, if that helps.

I didn't notice an official plugin infrastructure. Do you imagine it as just the same classes, same namespace, but living in their own `pay-contrib-polar` gem or something like that?

**Reply by excid3** [reply]
That should be fine. Everything would be namespaced under `Pay::Polar` or whatever.

If you catch any places where it'd be convenient to add a hook for plugins to integrate with, let me know and we can refactor. I am planning to do one for `Pay.sync` to handle checkout redirects back to the app syncing records and that'll need a way to add your own param handling.

**Comment by diegoquiroz** [comment]
Hey @mikker, did you abandon this? I'm interested in a Rails-Polar integration, let me know if you need a hand.

**Reply by mikker** [reply]
Ended up writing an API client for Polar but skipped Pay integration as pay was a bit too large and complex for my needs.

**Reply by diegoquiroz** [reply]
Yeah I saw the client you built, I'm actually using it right now but I saw this discussion and was curious if you got something more robust running. I think I'll keep using just the client. Thanks for developing it btw.
